### PR TITLE
A New Beneficial Viro Symptom: Neurological Overdrive

### PR DESCRIPTION
**Pull request by DatBoiTim**
## About The Pull Request
Adds a Symptom, Neurological Overdrive.

This is a level 9 beneficial, which provides stun reductions with the following thresholds:
Stage Speed 12: Faster Movement
Stage Speed 15: More stun reduction

## Why It's Good For The Game
Virology, currently, feels lacking for non-antagonistic behavior(It's fairly easy to make any malicious disease). Beneficials however, seem to be lacking in some way. Some of the obvious beneficials either kinda suck(Superficial, Regen Coma), others have weird side effects(Pituitary, Organic Flux), are annoying to others(Necro Seed, Cornu), or annoying in other ways(ThermoStable, Regen Coma).

This new Symptom I am hoping will add to the list of virology beneficials in a way that doesn't involve spreading pink shit everywhere.

## Changelog
:cl:
add: Added new Beneficial Virology Symptom, Neurological Overdrive.
/:cl:

### CODE REVIEW

**Comment by ItsTisme** [comment]
I think this ironcly still benefits antagonistic people more. Since security is the main force dealing stuns. Combat stuff like this is not really that helpfull for normal crew mainly for either sec or shitters/antags

**Comment by leestriter** [comment]
> I think this ironcly still benefits antagonistic people more. Since security is the main force dealing stuns. Combat stuff like this is not really that helpfull for normal crew mainly for either sec or shitters/antags

You are right this isn't something to make contagious or give to anyone, but it has its uses. Giving it to sec officers or giving it to people if armory is raided by an antag. It does have uses for an antag, sure, but they won't kill people with it.

**Comment by leestriter** [comment]
Needs some kinda downside like other symptoms, maybe increase hunger? Increased bleeding? Most symptoms have a set back, and while some of them are too severe, I think we should have a trade off here as well.

**Comment by leestriter** [comment]
Maybe give them extra burn vulnerability? So sure, its an anti stun, but you die to deadly lasers faster. Idk, just needs a tradeoff.

**Comment by DatBoiTim** [comment]
> Needs some kinda downside like other symptoms, maybe increase hunger? Increased bleeding? Most symptoms have a set back, and while some of them are too severe, I think we should have a trade off here as well.

Increased hunger results from faster move speed.

**Comment by DatBoiTim** [comment]
> Maybe give them extra burn vulnerability? So sure, its an anti stun, but you die to deadly lasers faster. Idk, just needs a tradeoff.

Burn vulnerability I could look at.

File: code/datums/diseases/advance/symptoms/neurological.dm [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+	stealth = 1
+	resistance = -1
+	stage_speed = 1
+	transmittable = -2
+	level = 9
+	severity = 0
+	symptom_delay_min = 1
+	symptom_delay_max = 1
+	var/fastmover = FALSE
+	var/list/fastmover_activation = list(FALSE, FALSE, FALSE, FALSE, FALSE) //Each of the Indexes Represents a Stage, When Activated, the Index is set to true
```
Comment (Archanial) [inline code/datums/diseases/advance/symptoms/neurological.dm]: what

Comment (Archanial) [reply code/datums/diseases/advance/symptoms/neurological.dm]: just make it int or something, `if stage > number then something` this looks terrible

Comment (DatBoiTim) [reply code/datums/diseases/advance/symptoms/neurological.dm]: Resolved by Commit: d31b2bb

**Comment by Fronsis** [comment]
Hulk 2.0 being born right here ?

**Comment by DatBoiTim** [comment]
> Hulk 2.0 being born right here ?

Viro already has a symptom which increases punch damage

Necropolis Seed provides a punch damage.

**Comment by eeSPee** [comment]
I mean, I like the idea, but this will only lead to greytide station wide.

Makes sec's tools and disablers a bit less useful. Stun reduction is not as significant cause bee doesn't have stuns anymore,

**Comment by leestriter** [comment]
> I mean, I like the idea, but this will only lead to greytide station wide.
> 
> Makes sec's tools and disablers a bit less useful. Stun reduction is not as significant cause bee doesn't have stuns anymore,

Yeah needs some sort of very significant trade off for sure.

File: code/datums/diseases/advance/symptoms/neurological.dm [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+	symptom_delay_min = 1
+	symptom_delay_max = 1
+	var/fastmover = FALSE
+	var/fastmover_activation = 0 //Each of the Indexes Represents a Stage, When Activated, the Index is set to true
+	var/super_reduction = FALSE
+	threshold_desc = "<b>Stage Speed 12:</b>The virus overclocks the nervous system making the host move even faster than normal.<br>\
+                      <b>Stage Speed 15:</b>The virus tightens the synapses between nerves, allowing for even quicker recovery from stuns."
+
+/datum/symptom/nerve_overdrive/OnAdd(datum/disease/advance/A)
+	A.infectable_biotypes |= MOB_ROBOTIC //Robots don't have nerves.
```
Comment (park66665) [inline code/datums/diseases/advance/symptoms/neurological.dm:19]: Manipulating `infectable_biotypes` should be kept at minimum. Most diseases don't have `MOB_ROBOTIC` in the first place, too.

Comment (PowerfulBacon) [reply code/datums/diseases/advance/symptoms/neurological.dm:19]: If robots dont have nerves why are you making robots susceptable?

**Review by PowerfulBacon — COMMENTED** [review]
This seems incredibly strong.
Also, does fast mover slow them down?

**Comment by DatBoiTim** [comment]
> This seems incredibly strong.
> 
> Also, does fast mover slow them down?

Yeah I'm working to add a burn vulnerability. 

And yes that slowdown is unintentional, I just forgot the -

File: code/datums/diseases/advance/symptoms/neurological.dm [hunk 3]
```diff
@@ -0,0 +1,81 @@
+/datum/symptom/nerve_overdrive
+	name = "Neurological Overdrive"
+	desc = "The disease increases connectivity between the infected's nerves, reducing stuns. Increased nerve connections however make burns far more painful."
```
Comment (zeskorion) [inline code/datums/diseases/advance/symptoms/neurological.dm:3]: Stun resistance is one thing I absolutely did not want to give virology, i dont want them able to just become entirely unstoppable in combat with a nice combat disease. Especially if you're adding a movement speed threshold here.

Comment (DatBoiTim) [reply code/datums/diseases/advance/symptoms/neurological.dm:3]: It's arguably worse than hulk or meth when mixed with mannitol. Your resistance to burn on each body part is reduced by 2.5. So even if you have Biometallic, you are at a net negative for burn resistance.

File: code/datums/diseases/advance/symptoms/neurological.dm (hunk 2 above)
Comment (DatBoiTim) [reply code/datums/diseases/advance/symptoms/neurological.dm:19]: Wait I thought that removed it.

**Comment by DatBoiTim** [comment]
After a bit of review this probably isn't a beneficial we want. This is just one of a series of me trying to add beneficials via throwing the spaghetti at the wall to see what sticks.

**Pull request closed without merging**